src/main/java/org/folio/circulation/resources/RequestCollectionResource.java

```diff
@@ -175,10 +182,15 @@ void get(RoutingContext routingContext) {
     final var clients = Clients.create(context, client);
 
     final var requestRepository = RequestRepository.using(clients);
-
     final var id = getRequestId(routingContext);
-
-    fromFutureResult(requestRepository.getById(id))
+    final var instanceRepository = new InstanceRepository(clients);
+    final var permissions = OkapiPermissions.from(context.getHeaders());
+    final var errorHandler = new OverridingErrorHandler(permissions);
+
+    fromFutureResult(requestRepository.getById(id)
+      .thenComposeAsync(r -> r.after(when(
+        req -> shouldFetchInstance(errorHandler, req),
+        req -> fetchInstance(instanceRepository, req), req -> ofAsync(() -> req)))))
```

Comment:
Should this be included in requestRepository.getById() ?

Reply:
Moved fetch to mentioned method, thank you.